datadog_checks_dev/datadog_checks/dev/tooling/test.py

```diff
@@ -0,0 +1,142 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from .constants import TESTABLE_FILE_EXTENSIONS, get_root
+from .git import files_changed
+from .utils import get_testable_checks
+from ..subprocess import run_command
+from ..utils import chdir, path_join
+
+STYLE_ENVS = {
+    'flake8',
+}
+
+
+def get_tox_envs(checks, style=False, benchmark=False, every=False, changed_only=False, sort=False):
+    testable_checks = get_testable_checks()
+    changed_checks = get_changed_checks() if changed_only else None
+
+    if not checks:
+        checks = sorted(
+            testable_checks & (
+                get_changed_checks() if changed_checks is None else changed_checks
```

Comment:
This is really confusing. You're always &-ing with `get_changed_checks()` because if `changed_check` is not `None` then it contains the result of `get_changed_checks()`

Reply:
Well then why don't we just have `changed_check = get_changed_checks()` ?

Reply:
Okay so if we do 
```python
changed_checks = get_changed_checks() if changed_only or not checks else None
```
then we should be able to just do
```python
if not checks:
    checks = sorted(testable_checks & changed_checks)
```
Right ? Just trying to simplify code here.

Reply:
Then line 34 would become

```
or ((changed_only or not checks) and check not in changed_checks)
```

which is worse

Reply:
well at this point `not checks` is always false since your in the `for check in checks` loop so line 34 doesn't need to change